hbase-protocol-shaded/src/main/protobuf/Master.proto

```diff
@@ -292,7 +292,8 @@ message IsInMaintenanceModeResponse {
 }
 
 message BalanceRequest {
-  optional bool force = 1;
+  optional bool ignore_rit = 1;
```

Comment:
Yes, the only thing that matters for protobuf is actually the ordinal -- not the position, but the number on the right hand side.

I renamed this so it would reflect the actual functionality of this argument. The current implementation of `force` only affects the region in transition behavior. The fact that it's called `force` is opaque and uninformative, and as part of the move to the more descriptive BalanceRequest it made sense to rename it.

If you'd prefer I leave it, I can.

Reply:
What this could break is if people have custom forks that are directly accessing the protobufs. I'm assuming the protobufs are InterfaceAudience.PRIVATE, not sure if that's explicitly documented somewhere. For anyone who's using the basic Admin interfaces, it will all work with the rename.

Let me know if you'd prefer me to leave it. It feels much more intuitive to call it what it is, rather than force. If we ever wanted to add a new functionality to force, I think we should add a new boolean instead and add it to the builder. So we'd have like a `ignore_maintenance_mode` and `setIgnoreMaintenanceMode()` on the BalanceRequest.

Reply:
>I renamed this so it would reflect the actual functionality of this argument. The current implementation of force only affects the region in transition behavior. The fact that it's called force is opaque and uninformative, and as part of the move to the more descriptive BalanceRequest it made sense to rename it.

I much prefer your new name, as names go.

> What this could break is if people have custom forks that are directly accessing the protobufs. I'm assuming the protobufs are InterfaceAudience.PRIVATE, not sure if that's explicitly documented somewhere.

Our documentation on [Aspirational Semantic Versioning](http://hbase.apache.org/book.html#hbase.versioning.post10) does not explicitly discuss the protobuf files themselves, only the wire compatibility between clients and servers. However, anywhere someone is working on a 3rd party client, they consume our protobuf files directly, and for those projects, this would be an incompatible change. _Perhaps_ this falls under the category of "HBase LimitedPrivate API", similar to coprocessor APIs, which allows us to make changes on minor releases.

I suggest starting a DISCUSS thread on the dev list, and filing a ticket to update our documentation according to the conclusion of that thread.

Nice one, Bryan.

Reply:
👍 I started the DISCUSS thread. For now I'll leave this as-is.